chore: update changelog for 0.3.1-rc (#479)
epiphany-platform/CommonDataLayer: CHANGELOG.md

# Common Data Layer Changelog

## 0.3.1-rc (Unreleased)
### Breaking changes

### New Components
- Object Builder (OB)
- Materializer - ondemand (M-OD)
- Materializer - general (M-G)
- CDL Web Admin (GUI)
- Edge Registry - (ER)

> - Partial Update Engine - (PUA)

Reviewer: PUA?

Reviewer 2: ```suggestion
- Partial Update Engine - (PUE)
```

### Changes to existing components:
- Schema Registry now uses Postgres Backend
- CDL can now work in ordered (messages) mode
- CDL now uses Open Telemetry

> - DataRouter supports Message Batching

Reviewer 2: ```suggestion
- Data Router supports Message Batching
```

### General Changes:
- Object Builder is available in Graphql API

> - ObjectBuilder supports empty filters

Reviewer 2: ```suggestion
- Object Builder supports empty filters
```

- Object Builder can use filters in queries
- Edge Registry is available in Graphql API
- Edge Registry notifications support added

> - All services can be queried with --help to get full list of options with explanations

Reviewer: I dunno if that will stay after I add config-files. In first version I've removed any trace of clap.

Author: yeah i remember but we will think about it when it gets merged

Reviewer: It got merged on Saturday

Reviewer: In theory it's not in `rc` as you've pushed tag before the merge, but for 0.3 `gold` it's pretty much in

### Major Bugfixes

> - DataRouter - It is now possible to swith to multi/single threaded behaviour

Reviewer 2: ```suggestion
- Data Router - It is now possible to swith to multi/single threaded behaviour
```

### Documentation:
- General Materialization documentation and how-to
- CDL versioning
- Updated ENV variables and their descriptions
- Add Feature List and descriptions

> - GRPC implementation

Reviewer 2: ```suggestion
- gRPC implementation
```

### General Fixes
- Consumer-based services will log error and continue on invalid message. no longer resulting in crash loop
- Query multiple returns error for timeseries
- build script fix
- Metrics port defaulted to invalid ones

### RFCs:
- Materialization RFC
- RFC index page
- partial materialization

> - edge repository

Reviewer 2: ```suggestion
- Edge Registry
```

- Initial draft of the MessagePack
- Alternative communication method to Kafka and RabbitMQ

## v0.2.0

### Breaking changes
- Archived document-storage crate
- Archived blob-store crate

### Features

> - Added store type of schema in schema registry

Reviewer 2: ```suggestion
- Added store type of schema in Schema Registry
```

- Added RAW queries to postgres

> - Added RAW to query-router

Reviewer 2: ```suggestion
- Added RAW to Query Router
```

- Added VictoriaMetrics Support

> - Added check if Kafka/AMQP topic exists before inserting new schema in schema registry

Reviewer 2: ```suggestion
- Added check if Kafka/AMQP topic exists before inserting new schema in Schema Registry
```

> - Added GRPC alternative ingestion method to command-service

Reviewer 2: ```suggestion
- Added gRPC alternative ingestion method to command-service
```
